gensim/models/tfidfmodel.py

```diff
@@ -39,24 +41,28 @@ def resolve_weights(smartirs):
 
     Returns
     -------
-    3-tuple (local_letter, global_letter, normalization_letter)
+    str of (local_letter, global_letter, normalization_letter) or None
 
     local_letter : str
         Term frequency weighing, one of:
-            * `n` - natural,
-            * `l` - logarithm,
+            * `b` - binary,
```

Comment:
In general, I would agree. However, these letters are standard, see [SMART Information Retrieval System](https://en.wikipedia.org/wiki/SMART_Information_Retrieval_System).

Reply:
I think the link to the Wiki page from our documentation would be helpful, but I still think replacing the magic strings with symbolic constants is worth it. My main motivation is to decrease the cognitive load on the reader and maintainer of the code. Even if the current magic strings are standard, you're expecting the reader/maintainer to remember what each one of them means in order to understand the code. This may not seem like much if you're very familiar with SMART, but it adds an unnecessary barrier to less enlightened people like myself.

Is there a compelling reason not to add symbolic constants? If you're worried about polluting the namespace, you could prefix each of them with `SMART_` to make it obvious what they are.

Reply:
> I think the link to the Wiki page from our documentation would be helpful

It is already part of the documentation.

> Is there a compelling reason not to add symbolic constants?

I see several reasons for opposing the addition of symbolic constants:

- The individual letters are clearly documented in the class docstring. No information is added by duplicating this information in the suggested form.
- Succint expressions such as `w_tf not in 'btnaldL'` become less readable in the suggested form.
- PR #1791 introduced the SMART letters. The PR was reviewed by two senior maintainers and found readable.

Reply:
> I think the link to the Wiki page from our documentation would be helpful

It is already part of the documentation.

> Is there a compelling reason not to add symbolic constants?

I see several reasons for opposing the addition of symbolic constants:

- The individual letters are clearly documented in the class docstring. No information is added by duplicating this information.
- Succint expressions such as `w_tf not in 'btnaldL'` become less readable in the suggested form.
- PR #1791 introduced the SMART letters. The PR was reviewed by two senior maintainers (Cc @piskvorky, @menshikh-iv) and found readable.

Reply:
@piskvorky and @menshikh-iv What do you think?

Reply:
I'm OK with direct string-like representation, as long as it's contained to the SMART module/functionality, and doesn't "magically" appear in other documentation / workflows.

That is, users who work explicitly with SMART can be expected to know the SMART nomenclature. But we shouldn't bother anyone else with it (users of plain TfIdf, Dictionary etc). As a user, I don't want to be studying some arcane lettering system just to use standard TFIDF.

I'll try to review the actual PR to see what's the case here, but that's my opinion in general.